clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -182,6 +183,32 @@ public void testInvalidSocketReceiveBufferSize() {
         new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
     }
 
+    @Test(expected = KafkaException.class)
+    public void testGlobalPoolWithNoPoolInstance() {
+        Map<String, Object> config = new HashMap<>();
+        config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        config.put(ConsumerConfig.POOL_CLASS_NAME_CONFIG, "org.apache.kafka.common.memory.GlobalPoolDelegate");
+        new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+    }
+
+    @Test
+    public void testKafkaConsumerGlobalMemoryPoolInstance() {
+        // Selector initialization calls memoryPool.size() to figure out low mem threshold.
+        // Using that to verify that the memory pool instance is delegated through GlobalPoolDelegate
+        MemoryPool mockMemoryPool = EasyMock.createMock(MemoryPool.class);
+        EasyMock.expect(mockMemoryPool.size()).andReturn(Long.MAX_VALUE).once();
+        EasyMock.replay(mockMemoryPool);
+
+        Map<String, Object> config = new HashMap<>();
+        config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        config.put(ConsumerConfig.POOL_CLASS_NAME_CONFIG, "org.apache.kafka.common.memory.GlobalPoolDelegate");
+        config.put("linkedin.memorypool.pool.instance", mockMemoryPool);
```

Comment:
should this be in consumer config as well? it looks a little strange that in addition to the POOL_CLASS_NAME_CONFIG, people will have to set another hard coded config in configs. Can we not do something similar to the key/value serializer class?